pyteal/ast/opup.py

```diff
@@ -87,35 +87,41 @@ def __init__(self, mode: OpUpMode, target_app_id: Expr = None):
 
         self.mode = mode
 
-    def _construct_itxn(self) -> Expr:
+    def _construct_itxn(self, inner_fee: Expr = None) -> Expr:
+        fields: dict[TxnField, Expr | list[Expr]] = {
+            TxnField.type_enum: TxnType.ApplicationCall
+        }
+
+        # If an inner_fee is specified, use it here
+        if inner_fee is not None:
+            require_type(inner_fee, TealType.uint64)
+            fields[TxnField.fee] = inner_fee
+
         if self.mode == OpUpMode.Explicit:
-            return Seq(
-                InnerTxnBuilder.Begin(),
-                InnerTxnBuilder.SetFields(
-                    {
-                        TxnField.type_enum: TxnType.ApplicationCall,
-                        TxnField.application_id: self.target_app_id,
-                    }
-                ),
-                InnerTxnBuilder.Submit(),
-            )
+            fields[TxnField.application_id] = self.target_app_id
         else:
-            return Seq(
-                InnerTxnBuilder.Begin(),
-                InnerTxnBuilder.SetFields(
-                    {
-                        TxnField.type_enum: TxnType.ApplicationCall,
-                        TxnField.on_completion: OnComplete.DeleteApplication,
-                        TxnField.approval_program: ON_CALL_APP,
-                        TxnField.clear_state_program: ON_CALL_APP,
-                    }
-                ),
-                InnerTxnBuilder.Submit(),
+            fields.update(
+                {
+                    TxnField.on_completion: OnComplete.DeleteApplication,
+                    TxnField.approval_program: ON_CALL_APP,
+                    TxnField.clear_state_program: ON_CALL_APP,
+                }
             )
 
-    def ensure_budget(self, required_budget: Expr) -> Expr:
+        return InnerTxnBuilder.Execute(fields)
+
+    def ensure_budget(self, required_budget: Expr, inner_fee: Expr = None) -> Expr:
         """Ensure that the budget will be at least the required_budget.
 
+        Args:
+            required_budget: minimum op-code budget to ensure for the
+                upcoming execution.
+            inner_fee (optional): OpUp utility requires inner transactions fee
+                to be paid by the upper call as default. To let the
+                application pay inner transactions fee with app account's
+                funds, set this parameter to `Global.min_txn_fee` or a given
+                fixed `Int`.
```

Comment:
I don't agree with this comment anymore. It should outline the behavior that will happen if you omit `inner_fee` (first any excess fee credit from the group is used, then the caller app's account is used if needed), and what will happen if you specify it. For the latter, there are two common cases:
1. (The most common and the reason this PR was created) If you specify zero for the fee, you ensure that the group's fee credit must pay for this inner txn
2. If you specify the min txn fee, you ensure that the app's account is used and the group's fee credit is unmodified

Also, it's very important to mention that _multiple_ inner txns may be created, and the specified fee applies to every one of them individually.